packages/pkh-webauthn/src/index.ts

```diff
@@ -0,0 +1,221 @@
+
+    const credentials = globalThis.navigator.credentials
+    const credential = await credentials.create(opts)
+    if (!credential) throw new Error('Empty Credential Response')
+    // @ts-ignore CredentialsContainer does contain response
+    const { response } = credential
+    const authenticatorData = getAuthenticatorData(response)
+    const { publicKey } = decodeAuthenticatorData(authenticatorData)
+    return encodeDIDFromPub(publicKey)
+
+  }
+
+  export async function getAuthMethod (didOpts: {
+    did?: string
+    dids?: Array<string>,
+    selectDID?: DIDSelector
+  }): Promise<AuthMethod> {
+    const { did, dids } = didOpts
+
+    let selectDID: DIDSelector
+
+    if (didOpts.selectDID) selectDID = didOpts.selectDID // Use callback
+    else if (did) selectDID = async () => did // Use known DID
+    else if (dids) selectDID = async (a, b) => dids.find(x => x == a || x == b) // Use probe result
+    else throw new Error('getAuthMethod({ did|dids|selectDID }) expects one resolution option')
+
+    return async (opts: AuthMethodOpts): Promise<Cacao> => {
+      return createCacao(opts, selectDID)
+    }
+  }
+
+  async function createCacao (opts: AuthMethodOpts, selectDID: DIDSelector): Promise<Cacao> {
+    const now = new Date()
+    // The public key is not known at pre-sign time.
+    // so we sign a "challenge"-block without Issuer attribute
+    const challenge: Cacao = {
+      h: {
+        t: 'caip122'
+      },
+      p: {
+        domain: globalThis.location.hostname,
+        aud: opts.uri || globalThis.location.toString(),
+        version: '1',
+        nonce: opts.nonce || u8a.toString(randomBytes(8), 'base64url'),
+        resources: opts.resources,
+        exp: opts.expirationTime || new Date(now.getTime() + 7 * 86400000).toISOString(), // 1 week
+        nbf: opts.notBefore || now.toISOString(),
+        iat: opts.issuedAt || now.toISOString()
+      } as Payload // squelch missing p.iss
+    }
+
+    const block = await blockFromCacao(challenge) // await CacaoBlock.fromCacao(challenge) when issue resolved
+    // perform sign
+    const { credential, recovered } = await authenticatorSign(block.cid.bytes)
+
+    // @ts-ignore - credential.response does exist.
+    const { response } = credential
+    const { clientDataJSON, signature } = response
+    const authData = getAuthenticatorData(response)
+    const aad = assertU8(encode({ authData, clientDataJSON }))
+
+    const recoveredDIDs = recovered.map(encodeDIDFromPub)
+
+    const iss = await selectDID(recoveredDIDs[0], recoveredDIDs[1])
+    if (!iss) throw new Error('PublicKeySelectionFailed')
+    // Assert that the resolved key belongs to the signature
+    if (!recoveredDIDs.includes(iss)) throw new Error('UnrelatedPublickey')
+
+    // Insert iss + aad after signature
+    return {
+      h: challenge.h,
+      p: {
+        ...challenge.p,
+        iss,
+      },
+      s: {
+        t: 'webauthn:p256',
+        s: u8a.toString(signature, 'base64url'),
```

Comment:
`s` should be Bytes, e.g. Uint8Array, according to [CACAO spec](https://chainagnostic.org/CAIPs/caip-74#specification)
```suggestion
        s: signature,
```

Reply:
I see, the type-definitions seem to be outdated compared to spec. `codecs/src/cacao.ts:51`.

I don't want to update the typedefs in this PR as it will most likely break static-analysis for modules that I am not familiar with. 
I think updating typedefs belongs in a separate issue; Changes applied but leaving a `@ts-ignore + TODO:` in `test/index.test.ts` for now.

Reply:
cc @ukstv Thoughts about tyhpedefs?

Reply:
@ukstv so the spec is wrong or we are wrong?

Reply:
@oed we are wrong.

Reply:
ugh, ok. This might be a problem if WalletConnect implements CAIP-222.

>> Looks closer

Actually in [CAIP-22 a string is also used](https://chainagnostic.org/CAIPs/caip-222#response), but it is a hex string 🙃

Reply:
Solana: base58btc-encoded, Ethereum: 0xHex, Tezos: tezos-specific signature string, that's what we have now.

Reply:
So, base64url (no pad) is okayish.

Reply:
Ok, let's use the base64url string then.

Reply:
@ukstv , @oed , Sorry for the delay, I needed to spend some time offline. 

I've patched the following:
- Reverted signature encoding back to base64url, tests pass.
- Added usage instruction to `src/indext.ts` following codebase praxis.

let me know if there is anything else i can do, otherwise hoping for a merge. Thanks